cpp/src/DataStorm/Instance.h

```diff
@@ -105,6 +99,37 @@ namespace DataStormI
 
         void destroy(bool);
 
+        // Helper methods to schedule and cancel timer tasks with the instance timers, schedule and cancel calls
+        // done after the instance is destroyed are ignored.
+
+        template<class Rep, class Period>
+        void scheduleTimerTask(std::function<void()> function, const std::chrono::duration<Rep, Period>& delay)
+        {
+            std::unique_lock<std::mutex> lock(_mutex);
```

Comment:
Should we throw an exception if the timer is null. Otherwise you get the illusion that it worked.

Reply:
These timers are used for reconnect sessions, or retry connection failures, there is a chance this happens during destroy and it is fine to ignore, the timer is only ever null if we are destroying the node.

we already throw invalid_argument from Ice::Timer in this case, another option would be to ignore it, but I think it is clear this way that we are only ignoring scheduled calls once we are destroyed.